kernel/syscall.c

```diff
@@ -1683,6 +1683,45 @@ long myst_syscall_readlink(const char* pathname, char* buf, size_t bufsiz)
     return ret;
 }
 
+long myst_syscall_readlinkat(
+    int dirfd,
+    const char* pathname,
+    char* buf,
+    size_t bufsiz)
+{
+    long ret = 0;
+    struct locals
+    {
+        char abspath[PATH_MAX];
+    };
+    struct locals* locals = NULL;
+
+    if (!pathname || !buf)
+        ERAISE(-EINVAL);
+
+    if (!(locals = malloc(sizeof(struct locals))))
+        ERAISE(-ENOMEM);
+
+    /* If pathname is absolute, then ignore dirfd */
+    if (*pathname == '/' || dirfd == AT_FDCWD)
+    {
+        ret = myst_syscall_readlink(pathname, buf, bufsiz);
+    }
+    else
+    {
+        ECHECK(_get_absolute_path_from_dirfd(
```

Comment:
This function is now renamed to `myst_get_absolute_path_from_dirfd`.

Reply:
Changed.